app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/importfiles/ImportFilesScreen.kt

```diff
@@ -57,31 +60,62 @@ private const val TOTAL_FILE_SIZE: Long = 50_000_000_000L
     closeActivity: () -> Unit,
     navigateToUploadProgress: () -> Unit,
 ) {
     val files by newTransferViewModel.importedFilesDebounced.collectAsStateWithLifecycle()
     val filesToImportCount by newTransferViewModel.filesToImportCount.collectAsStateWithLifecycle()
     val currentSessionFilesCount by newTransferViewModel.currentSessionFilesCount.collectAsStateWithLifecycle()
     val selectedTransferType by newTransferViewModel.selectedTransferType.collectAsStateWithLifecycle()
+    val validityPeriodState by newTransferViewModel.selectedValidityPeriodOption.collectAsStateWithLifecycle()
+    val downloadLimitState by newTransferViewModel.selectedDownloadLimitOption.collectAsStateWithLifecycle()
+    val passwordOptionState by newTransferViewModel.selectedPasswordOption.collectAsStateWithLifecycle()
+    val emailLanguageState by newTransferViewModel.selectedLanguageOption.collectAsStateWithLifecycle()
+    val appSettings by settingsViewModel.appSettingsFlow.collectAsStateWithLifecycle(null)
 
-    ImportFilesScreen(
-        files = { files },
-        filesToImportCount = { filesToImportCount },
-        currentSessionFilesCount = { currentSessionFilesCount },
-        selectedTransferType = GetSetCallbacks(
-            get = { selectedTransferType },
-            set = newTransferViewModel::selectTransferType,
-        ),
-        removeFileByUid = newTransferViewModel::removeFileByUid,
-        addFiles = newTransferViewModel::importFiles,
-        closeActivity = closeActivity,
-        navigateToUploadProgress = {
-            newTransferViewModel.sendTransfer()
-            navigateToUploadProgress()
-        },
-        initialShowUploadSourceChoiceBottomSheet = true,
-    )
+    fun onAdvancedOptionsValueSelected(option: SettingOption) {
+        when (option) {
+            is ValidityPeriodOption -> newTransferViewModel.selectTransferValidityPeriod(option)
+            is DownloadLimitOption -> newTransferViewModel.selectTransferDownloadLimit(option)
+            is PasswordTransferOption -> newTransferViewModel.selectTransferPasswordOption(option)
+            is EmailLanguageOption -> newTransferViewModel.selectTransferLanguage(option)
+        }
+    }
+
+    appSettings?.let { safeAppSettings ->
+        newTransferViewModel.initTransferAdvancedOptionsValues(safeAppSettings)
+
+        val advancedOptionsCallbacks = AdvancedOptionsCallbacks(
+            initialValues = SettingsViewModel.AppSettingsData(
+                validityPeriod = safeAppSettings.validityPeriod,
+                downloadLimit = safeAppSettings.downloadLimit,
+                passwordOption = PasswordTransferOption.NONE,
+                emailLanguage = safeAppSettings.emailLanguage,
+            ),
+            advancedOptionsStates = { listOf(validityPeriodState, downloadLimitState, passwordOptionState, emailLanguageState) },
+            onAdvancedOptionsValueSelected = ::onAdvancedOptionsValueSelected,
+        )
```

Comment:
Right now, everytime the appSettings' state changes, this callback will be recreated because of the "initialValues". This will later recompose all of the screen because it's not a state. To make sure on the correct approach to take, maybe we can talk it out with @sirambd instead of me writing out alternatives I have a few doubts on. Because I don't what know what's the best way to avoid having too much recompositions when only certain parts of the callback have changed.

Reply:
Also initialValues and advancedOptionsStates kind of holds the same data, I feel like there should be a way to only have one does it not? Have you already considered this option?

Reply:
Maybe "initialValues" should be called something else outside of the bottom sheet because outside of the bottom sheet this value simply represents the current setting value.

Reply:
The AppSettings can't be modified during this screen

Reply:
We merged the usage of initialValues and states